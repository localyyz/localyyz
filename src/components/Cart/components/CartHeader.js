import React from "react";
import { View, StyleSheet, Text } from "react-native";
import { Sizes, Styles } from "localyyz/constants";

// third party
import PropTypes from "prop-types";

export default class CartHeader extends React.Component {
  static propTypes = {
    title: PropTypes.string,
    children: PropTypes.node,
    icon: PropTypes.node
  };

  static defaultProps = {
    title: ""
  };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>{this.props.title}</Text>
        <View style={styles.content}>
          <Text style={styles.contentLabel}>{this.props.children}</Text>
        </View>
        {this.props.icon}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    ...Styles.Horizontal,
    flex: 1,
    marginHorizontal: Sizes.InnerFrame,
    marginVertical: Sizes.InnerFrame,
    alignItems: "center"
  },

  title: {
    ...Styles.Text,
    ...Styles.Emphasized
  },

  content: {
    ...Styles.Horizontal,
    flex: 1,
    marginHorizontal: Sizes.InnerFrame / 2
  },

  contentLabel: {
    ...Styles.Text
  }
});
