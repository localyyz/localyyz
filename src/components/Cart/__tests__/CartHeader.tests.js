import React from "react"
import CartHeader from "../components/CartHeader"
import renderer from "react-test-renderer"

describe("CartHeader", () => {
    it("CartHeader: should render properly", () => {
        let props: {
            icon: "3d-rotation"
        }
        let rendered = renderer.create(<CartHeader {...props}/>);
        expect(rendered).toMatchSnapshot()
    })
})