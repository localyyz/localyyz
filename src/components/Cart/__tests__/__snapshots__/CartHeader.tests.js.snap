// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartHeader CartHeader: should render properly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "marginHorizontal": 15,
      "marginVertical": 15,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0)",
        "color": "#000000",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginHorizontal": 7.5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "color": "rgb(30, 30, 30)",
          "fontSize": 16,
          "fontWeight": "300",
        }
      }
    />
  </View>
</View>
`;
